apps/backend/src/modules/notifications/notification.controller.ts

```diff
@@ -66,19 +83,28 @@ class NotificationController extends BaseController {
 	 *                   type: array
 	 *                   items:
 	 *                     $ref: "#/components/schemas/Notification"
+	 *                 totalItems:
+	 *                   type: integer
+	 *                   description: The total number of notifications
 	 */
-	private async findAll(
-		options: APIHandlerOptions,
-	): Promise<APIHandlerResponse> {
-		const { user } = options;
 
+	private async findAll({
+		query,
+		user,
+	}: APIHandlerOptions<{
+		query: NotificationGetAllRequestDto;
+	}>): Promise<APIHandlerResponse> {
+		const { page, pageSize } = query;
 		const typedUser = user as { id: number };
 
 		return {
-			payload: await this.notificationService.findAll(typedUser.id),
+			payload: await this.notificationService.findAll({
+				page,
+				pageSize,
+				userId: typedUser.id,
```

Comment:
let's avoid variables just for typescript, where we can. Also we have a specific type for user in request
```suggestion
				userId: (user as UserAuthResponseDto).id,
```

Reply:
I resolved the conversation, but https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/490#discussion_r1771722196, recommended for this way instead.